src/d3a/d3a_core/sim_results/plotly_graph.py

```diff
@@ -119,29 +122,67 @@ def modify_time_axis(data: dict, title: str):
         for di in range(len(data)):
             time_list = data[di]["x"]
             for ti in time_list:
-                day_set.add(pendulum.datetime(ti.year, ti.month, ti.day, tz=TIME_ZONE))
+                day_set.add(
+                    pendulum.datetime(ti.year, ti.month, ti.day, ti.hour, ti.minute, tz=TIME_ZONE)
+                )
 
         day_list = sorted(list(day_set))
         if len(day_list) == 0:
             raise ValueError("There is no time information in plot {}".format(title))
 
-        start_time = pendulum.datetime(day_list[0].year, day_list[0].month, day_list[0].day,
-                                       0, 0, 0, tz=TIME_ZONE)
-        end_time = pendulum.datetime(day_list[-1].year, day_list[-1].month, day_list[-1].day,
-                                     23, 59, 59, tz=TIME_ZONE)
+        start_time = pendulum.datetime(
+            day_list[0].year, day_list[0].month, day_list[0].day,
+            day_list[0].hour, day_list[0].minute, day_list[0].second, tz=TIME_ZONE
+        )
+        end_time = pendulum.datetime(
+            day_list[-1].year, day_list[-1].month, day_list[-1].day,
+            day_list[-1].hour, day_list[-1].minute, day_list[-1].second, tz=TIME_ZONE)
 
         return [start_time, end_time], data
 
+    @classmethod
+    def plot_slider_graph(cls, fig, stats_plot_dir, area_name, market_slot_data_mapping):
+        steps = []
+        for i in range(len(market_slot_data_mapping)):
+            step = dict(
+                method="update",
+                args=[{"visible": [False] * len(fig.data)},
+                      {"title": "Slider switched to slot: " + str(i)}],  # layout attribute
+            )
+            for k in range(market_slot_data_mapping[i].start,
+                           market_slot_data_mapping[i].end):
+                step["args"][0]["visible"][k] = True  # Toggle i'th trace to "visible"
+            steps.append(step)
+        sliders = [dict(
+            active=0,
+            currentvalue={"prefix": "MarketSlot: "},
+            pad={"t": len(market_slot_data_mapping)},
+            steps=steps
+        )]
+        output_file = os.path.join(stats_plot_dir, f"offer_bid_trade_history.html")
```

Comment:
please include the market name in the filename. thanks!

Reply:
No need as im getting rid of un necessary nesting of additional folder